drivers/wifi/Kconfig.winc1500

```diff
@@ -0,0 +1,62 @@
+# Kconfig - Atmel WINC1500 WiFi driver options
+
+#
+# Copyright (c) 2015 Intel Corporation
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+menuconfig WIFI_WINC1500
+	bool "WINC1500 driver support"
+	default n
+	select ATMEL_WINC1500
+
+if WIFI_WINC1500
+
+config WIFI_WINC1500_NAME
+	string "Driver name"
+	default "WINC1500"
+
+config WIFI_WINC1500_SPI_DRV_NAME
+	string "SPI device where WINC1500 is connected"
+	default "SPI_1"
+	help
+	 Specify the device name of the SPI device to which WINC1500 is
+	 connected.
+
+config WIFI_WINC1500_SPI_SLAVE
+	int "SPI Slave Select where WINC1500 is connected"
+	default 1
+	help
+	 Specify the slave select pin of the SPI to which WINC1500 is
+	 connected.
+
+config WIFI_WINC1500_SPI_FREQ
+	int "SPI frequency to use with WINC1500"
+	default 4
+	help
+	 SPI frequency to use with WINC1500
+
+# TODO GPIO
+
+config WIFI_WINC1500_SSID
+	string "SSID the chip shall try to connect to"
+	default ""
+	help
+	 Set the SSID you want the chip to connect to. Note: only WPA PSK
+	 based AP is supported. Test purpose only, a full featured interface
+	 will be used at some point for more flexible configuration.
+
+config WIFI_WINC1500_SSID_LENGTH
+	int "Length of the SSID (do not count the \0)"
+	default 0
+	help
+	 SSID length, without counting the terminal character '\0'
+
+config WIFI_WINC1500_PSK
```

Comment:
So would the actual password be in the project's prj.conf file?  My experience placing passwords in source files, is they have a way of accidentally being pushed up in commits.   I've had this problem in the past, so decided to pass the password and SSID on the build command line, and/or place in local environment variables.
Would the method, as used in this PR: #5028, be perhaps safer?

Reply:
Current PR is just the bootstrap of winc1500, which code was primarily made by someone else.

Thing is, inserting fixed credential, as pre-provisioning, at build time or via Kconfig is same: it sucks. 

Btw PR #5028 has a flaw, SSID are not character arrays.

Future patches will introduce a proper wifi net mgmt interface, that all wifi drivers will have to follow.
Such interface could be properly used via a generic pre-provisioning solution via some flash somewhere etc...